test_tableslurp.sh

```diff
@@ -0,0 +1,142 @@
+#! /bin/sh
+
+# Script Name: test_tableslurp.sh
+
+# Author: Kim Lew
+
+# -- Various checks before testing of file(s) retrieval(s).
+# DIR_LOCAL is: /tmp/tablesnap-test
+# BUCKET=$(USER)-tablesnap-tests
+PYTHON_VERSION=$1
+DIR_LOCAL=$2
+AWS_ACC_KEY_ID=$3
+AWS_SEC_ACC_KEY=$4
+BUCKET=$5
+
+# Check that correct number of arguments were passed in from the Makefile
+# when running this script. If incorrect number, give a general error.
+echo
+echo "# args passed: $#"
+if [ "$#" -ne 5 ]; then
+	echo "5 parameters are required & some are missing. Check Makefile."
+fi
+
+# Check for which Python string was passed in as an argument & assign
+# corresponding Python environment directory.
+if [ "$PYTHON_VERSION" = "python2" ]; then
+  PYTHON_ENV="py27env"
+else # $1 = python3
+  PYTHON_ENV="py36env"
+fi
+
+# The directory, /tmp/tablesnap-test, should already exist since it is created
+# in the Makefile. If it does not exist, this script creates it.
+[ -d "$DIR_LOCAL" ] || mkdir -p "$DIR_LOCAL"
+
+# Check if AWS CLI is in container & if not, install it.
+if ! aws --version > /dev/null; then
+  echo "Error: You are missing the aws cli program."
+  echo "Install AWS CLI version 2."
+  echo "Then re-run this script."
+  exit 1
+fi
+
+# Set AWS_ACCESS_KEY_ID & AWS_SECRET_ACCESS_KEY as environment variables. Then,
+# use aws cli command, to check if moved or written file is in the container.
+export AWS_ACCESS_KEY_ID="$AWS_ACC_KEY_ID"
+export AWS_SECRET_ACCESS_KEY="$AWS_SEC_ACC_KEY"
+
+# Create 2 files with touch. Add test file location, i.e., file path & file name
+# part into index file as content & copy the 2 files up to S3. Example files:
+# /tmp/tablesnap-test/test_aws_s3_cp.txt
+# /tmp/tablesnap-test/index.txt
+TEST_FILE="test_aws_s3_cp.txt"
+INDEX_FILE="index.txt"
+RENAMED_FILE="test_file_orig.txt"
+CONTAINER="tester"
+SRC_LOCAL="${DIR_LOCAL}/${TEST_FILE}"
+INDEX_SRC="${DIR_LOCAL}/${INDEX_FILE}"
+INDEX_BUCKET="${BUCKET}"
+BACKUP_BUCKET="${BUCKET}"
+
+# Create destinations on S3. Note: Dest for index file is at top-level,
+# i.e., not under a directory.
+# s3://kimlew-tablesnap-tests/tester:/tmp/tablesnap-test/test_aws_s3_cp.txt
+DEST_ON_S3=s3://"${BUCKET}"/"${CONTAINER}":"${DIR_LOCAL}"/"${TEST_FILE}"
+INDEX_DEST=s3://"${BUCKET}"/"${CONTAINER}":/"${INDEX_FILE}"
+
+echo "SRC_LOCAL is:" "${SRC_LOCAL}"
+echo "DEST_ON_S3 is:" "${DEST_ON_S3}"
+echo "INDEX_SRC is:" "${INDEX_SRC}"
+echo "INDEX_DEST is:" "${INDEX_DEST}"
+echo
+
+touch "${SRC_LOCAL}" > /dev/null
```

Comment:
Suggestion: instead of using `touch` to create the `$SRC_LOCAL` file, redirect the output of `date` into the file:

```
date > "${SRC_LOCAL}"
```

This would allow us to make a more interesting test later on, where we can compare the content of the file that `tableslurp` downloads to this file (after it's renamed)

Reply:
Good suggestion. Changed.

Reply:
Confirmed using `date` to give the file some contents.